app/src/main/java/com/github/sdp_begreen/begreen/firebase/FirebaseDB.kt

```diff
@@ -110,14 +115,27 @@ object FirebaseDB: DB {
         return storePicture(image, USER_PROFILE_PICTURE_METADATA, metadata,
             databaseReference.child(USERS_PATH).child(userId),
             storageReference.child(USERS_PATH).child(userId).child(
-                USER_PROFILE_PICTURE_METADATA))
+                USER_PROFILE_PICTURE_METADATA)) as ProfilePhotoMetadata
     }
 
-    override suspend fun addImage(image: Bitmap, userId: Int, metadata: PhotoMetadata): PhotoMetadata? {
+    override suspend fun addTrashPhoto(image : Bitmap, trashPhotoMetadata: TrashPhotoMetadata): TrashPhotoMetadata? {
 
-        return storePicture(image, null, metadata,
-            databaseReference.child("pictures").child(userId.toString()),
-            storageReference.child("userId").child(userId.toString()))
+        checkArgument(
+            !trashPhotoMetadata.takenBy.isNullOrBlank(),
+            "The user that took the photo cannot be blank or null"
+        )
+
+        checkArgument(
+            userExists(trashPhotoMetadata.takenBy!!),
+            "The user doesn't exist in the database"
+        )
+
+        var newPhotoMetadata = trashPhotoMetadata.copy(pictureId = null)
+
+
+        return storePicture(image, USER_POSTS, newPhotoMetadata,
+            databaseReference.child(USERS_PATH).child(newPhotoMetadata.takenBy!!).child(USER_POSTS),
+            storageReference.child(USERS_PATH).child(newPhotoMetadata.takenBy!!).child(USER_POSTS)) as TrashPhotoMetadata

```

Comment:
Made the change !

Reply:
Actually, after testing it seems that the cast is necessary. I suppose this is due to the fact that `storePicture` returns a generic `T` type. The data are not lost with the generic type but we still should precise to `addTrashPhoto` that the returned object is indeed a `TrashPhotoMetadata`. Otherwise we could set the returning type of `addTrashPhoto` to `T` but we would have to cast it each time we call the function to usure that it's indeed the expected type.

Reply:
I am really not convinced by this explanation, it doesn't really make sens to me. If we return a type `T` and this type is a `TrashPhotoMetadata` then the returned type is also `TrashPhotoMetadata`. 

But as we have to move forward with this PR in order to be able to continue working since it changed pretty much the whole project, I will accept this PR as is.